litestar/datastructures/headers.py

```diff
@@ -380,15 +405,15 @@ def from_header(cls, header_value: str) -> "ETag":
         weak, value = match.group(1, 2)
         try:
             return cls(weak=bool(weak), value=value)
-        except ValidationError as exc:
+        except ValueError as exc:
             raise ImproperlyConfiguredException from exc
 
-    @validator("value", always=True)
-    def validate_value(cls, value: Any, values: Dict[str, Any]) -> Any:
-        """Ensure that either value is set or the instance is for ``documentation_only``."""
-        if values.get("documentation_only") or value is not None:
-            return value
-        raise ValueError("value must be set if documentation_only is false")
+    def __post_init__(self) -> None:
+        pattern = re.compile(r"^[ -~]+$")
```

Comment:
The benefit of per-compiling regex patterns is lost if you do it on every instantiation like this. Either move it into the class scope with `ClassVar` or into the global scope and reference it here.